lib/manageiq/providers/workflows/builtin_methods.rb

```diff
@@ -11,15 +11,53 @@ def self.email(params = {}, _secrets = {}, _context = {})
         end
 
         private_class_method def self.email_status!(runner_context)
-          miq_task = ::MiqTask.find(runner_context["miq_task_id"])
-          return if miq_task.nil?
+          miq_task_status!(runner_context)
+        end
+
+        def self.embedded_ansible(params = {}, _secrets = {}, _context = {})
+          repository_url, repository_branch, playbook = params.values_at("RepositoryUrl", "RepositoryBranch", "Playbook")
+
+          vars = params
+                 .slice("Hosts", "ExtraVars", "BecomeEnabled", "Timeout", "Verbosity", "Credential", "CloudCredential", "NetworkCredential", "VaultCredential")
+                 .transform_keys { |k| k.underscore.to_sym }
+
+          repository = ::ConfigurationScriptSource.find_by(:scm_url => repository_url, :scm_branch => repository_branch)
+          return error!(runner_context, :cause => "Unable to find Repository: URL: [#{repository_url}] Branch: [#{repository_branch}]") if repository.nil?
+
+          playbook = ::ConfigurationScriptPayload.find_by(:configuration_script_source => repository, :name => playbook)
+          return error!(runner_context, :cause => "Unable to find Playbook: Name: [#{playbook}] Repository: #{repository.name}") if playbook.nil?
```

Comment:
NOTE I thought that looking up a playbook by repository_url/repository_branch+playbook_path was the most unique way to identify a playbook without getting into IDs which obviously wouldn't work across multiple miq instances.

Reply:
I wonder if we can support both, but yeah I agree this is a good way to start it.

Reply:
Yeah I was thinking of adding a PlaybookId and you have to specify either PlaybookId or RepoUrl+RepoBranch+PlaybookPath

Reply:
The non ID way would be better for service import/export as well